### PR TITLE
 Refine SbPlayer tests for multiple videos

### PR DESCRIPTION
**Pull request by WeiChungChang**
[nplb] Refine SbPlayer tests for multiple videos

b/230877781

### CODE REVIEW

File: starboard/nplb/multiple_player_test.cc [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+namespace starboard {
+namespace nplb {
+namespace {
+
+using ::starboard::testing::FakeGraphicsContextProvider;
+using ::testing::ValuesIn;
+
+typedef SbPlayerTestFixture::GroupedSamples GroupedSamples;
+typedef std::vector<SbPlayerTestConfig> SbPlayerMultiplePlayerTestConfig;
```
[the remark this answers is not in the document]
Comment (WeiChungChang) [reply starboard/nplb/multiple_player_test.cc:31]: https://lbshell-internal-review.git.corp.google.com/c/cobalt_src/+/242602/comment/3d12baf5_e64eabc8/

Please refer to previous comment regarding the use of a map and consider changing it to a vector (without run-length coding). 

I am open to either using a map or a vector, but I need clear guidance without any contradictions. Kindly provide your suggestion on this matter. Thank you.

File: starboard/nplb/multiple_player_test.cc [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+ public:
+  PlayerThread(const SbPlayerTestConfig& config,
+               FakeGraphicsContextProvider* fake_graphics_context_provider,
+               const std::function<void(SbPlayerTestFixture*)>& functor)
+      : config_(config),
+        functor_(functor),
+        fake_graphics_context_provider_(fake_graphics_context_provider) {}
+
+  void Run() override {
+    SbPlayerTestFixture player(config_, fake_graphics_context_provider_);
```
Comment (jasonzhangxx) [inline starboard/nplb/multiple_player_test.cc:153]: What's the consideration to initialize the |player| out of the |functor_|?

File: starboard/nplb/multiple_player_test.cc [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    int samples_to_write = SbPlayerGetMaximumNumberOfSamplesPerWrite(
+        player_fixture->GetPlayer(), kSbMediaTypeVideo);
+    samples.AddVideoSamplesWithEOS(0, samples_to_write);
+  }
+
+  ASSERT_NO_FATAL_FAILURE(player_fixture->Write(samples));
+  ASSERT_NO_FATAL_FAILURE(player_fixture->WaitForPlayerEndOfStream());
+}
+
+void WriteMultipleBatches(SbPlayerTestFixture* player_fixture) {
```
[the remark this answers is not in the document]
Comment (WeiChungChang) [reply starboard/nplb/multiple_player_test.cc:112]: The function comes from CL 232041's review comments.

The instruction directs me to create a test pattern, e.g. given a maximum batch size of 8. The testing process involves writing test batches starting from 8 and decreasing by one until either the testing is complete or there are no more test samples available,as 8, 7, 6 ... 1.

Starting from the original test pattern, it possesses its own representation that allows for exploration and verification of the entire range [1, ... max_sample_to_write] to ensure the legality of different samples to write.

1. The syntax has undergone a complete change, leaving me unaware of the root cause behind the issue."

2. The test pattern has been constructed according to the provided description in review comments.

**Comment by WeiChungChang** [comment]
no action

**Pull request closed without merging**